### ISSUE TITLE
CheckboxGroup - adding checkbox as helper component breaks the checkboxgroup logic

### ISSUE THREAD

**Issue by nSharifA** [vaadin/flow-components#3596]
### Description

I wanted to add a selectAll/deselectAll checkbox to checkboxgroup's helper part (see the example).
Selecting the selectAll checkbox fires a value change event for checkboxGroup.

### Expected outcome

I know the setHelperComponent method was not created for this purpose, but the CheckboxGroup should not fire a value change event. 

### Minimal reproducible example

```
    ListDataProvider<String> dataProvider = new ListDataProvider<>(Arrays.asList("1", "2", "3"));
    
    CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
    checkboxGroup.setDataProvider(dataProvider);
    add(checkboxGroup);
    
    Checkbox selectAll = new Checkbox();
    selectAll.setLabelAsHtml("<i>Select/Deselect all</i>");
    checkboxGroup.setHelperComponent(selectAll);
    
    checkboxGroup.addValueChangeListener(e -> {
      System.out.print(e.getValue().size()); 
    });
```

### Steps to reproduce

1. Add the snippet to a View
2. Click on the checkbox called "1" -> expected value on console '1'
3. Click on the checkbox called "2" -> expected value on console '12'
4. Click on the checkbox called "3" -> expected value on console '123'
5. Click on the checkbox called "Select/Deselect all" -> expected value on console '123', instead '1234'

### Environment

Vaadin version(s): 14.8.11
OS: Windows 10

### Browsers

Issue is not browser related

**Comment by knoobie** [comment]
Interesting, from a technical point of view the problem this is really a miss-use of the helper but the output is kinda interesting.. 

to enhance the example:

```java
    checkboxGroup.addValueChangeListener(e -> {
      e.getValue().forEach(s -> {
        System.out.println("Value selected: " + s);
      });
      System.out.println("-----");
    });
```

After clicking `1`, `2`, `3` and `Select/Deselect All` it results in: 

```java
Value selected: 1
-----
Value selected: 1
Value selected: 2
-----
Value selected: 1
Value selected: 2
Value selected: 3
-----
Value selected: null
Value selected: 1
Value selected: 2
Value selected: 3
-----
```

It's probably related to the fact that `vaadin-checkbox` is just declared as `slot=helper` and for `vaadin-checkbox-group` it's just another checkbox. Luckily the server doesn't allow invalid values (here Select/Deselect All).

**Comment by web-padawan** [comment]
> this is really a miss-use of the helper

Yes, this is not a supported use case and we never considered it.
Maybe we could show a warning when using checkbox as a helper component.

**Comment by yuriy-fix** [comment]
We could generalise the documentation enhancement by saying that input components are not supposed to be used within helper.

**Comment by yuriy-fix** [comment]
Dear @nSharifA, could you please try wrapping the checkbox within `div` as a workaround for you particular case?
